app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt

```diff
@@ -7,28 +7,49 @@ import org.p2p.wallet.swap.ui.jupiter.tokens.SwapTokensContract
 import org.p2p.wallet.swap.ui.jupiter.tokens.interactor.SwapTokensInteractor
 
 class SwapTokensPresenter(
-    private val tokenToChange: SwapTokensChangeToken,
+    private val tokenToChange: SwapTokensListMode,
     private val mapper: SwapTokensToCellItemsMapper,
     private val interactor: SwapTokensInteractor,
 ) : BasePresenter<SwapTokensContract.View>(), SwapTokensContract.Presenter {
 
+    private var allTokens: List<SwapTokenModel> = emptyList()
+    private var currentToken: SwapTokenModel? = null
+
     override fun attach(view: SwapTokensContract.View) {
         super.attach(view)
 
         launch {
             val currentTokenToSwapTokens = when (tokenToChange) {
-                SwapTokensChangeToken.TOKEN_A -> interactor.getCurrentTokenA() to interactor.getAllTokensA()
-                else -> return@launch
+                SwapTokensListMode.TOKEN_A -> interactor.getCurrentTokenA() to interactor.getAllTokensA()
+                SwapTokensListMode.TOKEN_B -> interactor.getCurrentTokenB() to interactor.getAllAvailableTokensB()
             }
-            val cellItems = mapper.toCellItems(
-                chosenToken = currentTokenToSwapTokens.first,
-                swapTokens = currentTokenToSwapTokens.second
-            )
-            view.setTokenItems(cellItems)
+            currentToken = currentTokenToSwapTokens.first
+            allTokens = currentTokenToSwapTokens.second
+            renderAllSwapTokensList(allTokens)
         }
     }
 
+    private fun renderAllSwapTokensList(tokens: List<SwapTokenModel>) {
+        val cellItems = mapper.toCellItems(
+            chosenToken = currentToken ?: return,
+            swapTokens = tokens
+        )
+        view?.setTokenItems(cellItems)
+    }
+
     override fun onSearchTokenQueryChanged(newQuery: String) {
+        if (newQuery.isBlank()) {
+            renderAllSwapTokensList(allTokens)
+        } else {
+            launch {
```

Comment:
maybe you'll move it inside `renderSearchTokenList`?

Reply:
i'll check, thanks